Add a Godot script verification step to CICD

**What**

As discovered in #4 there were issues with checking the scripts using Godot in Linux. At the time I thought the issue was with Linux specifically, but after some new experimentation I've discovered that the issue is most likely due to Godot not knowing where each script/class is located.

Within the `.godot` folder there is a file called `global_script_class_cache.cfg` that seems to track information about custom classes. Create a new branch to test if a script check with Godot could be implemented if this file was added to source control.

**Why**

Tools provided in `gdtoolkit` can't catch logical errors in the code. That is to say, it will not raise errors when accessing variables, functions, or classes that haven't been declared and defined, and that is a crucial check that we want to be a part of the CICD pipeline.

Unfortunately still running into issues with 4.4-dev-7, even though in a local container everything works after the exact same steps.

The import project runs into the same errors, but in a local container this causes no issues. In the GitHub Actions container however the script is unable to resolve Ikaros, even when it is just a static class.

Import project:
```
Run godot --headless --quit --import
Godot Engine v4.4.dev.custom_build.46c8f8c5c (2024-12-18 17:23:54 UTC) - https://godotengine.org/

ERROR: Condition "!unique_ids.has(p_id)" is true. Returning: String()
   at: get_id_path (core/io/resource_uid.cpp:155)
ERROR: Condition "!unique_ids.has(p_id)" is true. Returning: String()
   at: get_id_path (core/io/resource_uid.cpp:155)
ERROR: Condition "!unique_ids.has(p_id)" is true. Returning: String()
   at: get_id_path (core/io/resource_uid.cpp:155)
ERROR: Condition "!unique_ids.has(p_id)" is true. Returning: String()
   at: get_id_path (core/io/resource_uid.cpp:155)
ERROR: Condition "!unique_ids.has(p_id)" is true. Returning: String()
   at: get_id_path (core/io/resource_uid.cpp:155)
ERROR: Do not use progress dialog (task) while flushing the message queue or using call_deferred()!
   at: add_task (editor/progress_dialog.cpp:169)
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR:  Config File "user://gut_temp_directory/gut_editor_config.json" does not exist.
ERROR: Condition "!tasks.has(p_task)" is true. Returning: canceled
   at: task_step (editor/progress_dialog.cpp:203)
ERROR: Condition "!tasks.has(p_task)" is true.
   at: end_task (editor/progress_dialog.cpp:226)
```

Verify engine version:
```
Run godot --headless --quit --script Ikaros/scripts/verify_engine_version.gd
Godot Engine v4.4.dev.custom_build.46c8f8c5c (2024-12-18 17:23:54 UTC) - https://godotengine.org/

SCRIPT ERROR: Parse Error: Identifier "Ikaros" not declared in the current scope.
          at: GDScript::reload (res://Ikaros/scripts/verify_engine_version.gd:11)
SCRIPT ERROR: Parse Error: Identifier "Ikaros" not declared in the current scope.
          at: GDScript::reload (res://Ikaros/scripts/verify_engine_version.gd:12)
SCRIPT ERROR: Parse Error: Identifier "Ikaros" not declared in the current scope.
          at: GDScript::reload (res://Ikaros/scripts/verify_engine_version.gd:14)
ERROR: Failed to load script "res://Ikaros/scripts/verify_engine_version.gd" with error "Parse error".
   at: load (modules/gdscript/gdscript.cpp:3040)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/godot-ci.yml
+++ b/.github/workflows/godot-ci.yml
@@ -1,6 +1,10 @@
 name: "godot-ci"
 on: push
 
+defaults:
+  run:
+    working-directory: ./Ikaros
+
 jobs:
   gdtoolkit:
     name: "gdtoolkit: gdformat and gdlint"
@@ -8,9 +12,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: Scony/godot-gdscript-toolkit@master
-      - run: gdformat --check Ikaros/core Ikaros/game
+      - run: gdformat --check core game
         if: always()
-      - run: gdlint Ikaros/core Ikaros/game
+      - run: gdlint core game
         if: always()
   godot-check-scripts:
     name: "godot: check scripts"
@@ -23,7 +27,6 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - name: Import project
-        working-directory: ./Ikaros
         run: godot --headless --quit --import
       - name: Verify engine version
-        run: godot --headless --quit --script Ikaros/scripts/verify_engine_version.gd
+        run: godot --headless --quit --script scripts/verify_engine_version.gd

